charts.py

```diff
@@ -0,0 +1,230 @@
+    if current_pair == stable_coin_pair:
+        for stable_coin in src.settings.DEBT_TOKENS[:-1]:
+            debt_token = stable_coin
+            main_chart_data, collateral_token_price = app.process_liquidity(
+                main_chart_data, collateral_token, debt_token
+            )
+    else:
+        main_chart_data, collateral_token_price = app.process_liquidity(
+            main_chart_data, collateral_token, debt_token
+        )
+    return main_chart_data, collateral_token_price
+
+
+def get_usd_debt_boundaries(col1, streamlit, loans_data):
+    with col1:
+        debt_usd_lower_bound, debt_usd_upper_bound = streamlit.slider(
+            label="Select range of USD borrowings",
+            min_value=0,
+            max_value=int(loans_data["Debt (USD)"].max()),
+            value=(0, int(loans_data["Debt (USD)"].max())),
+        )
+    streamlit.dataframe(
+        loans_data[
+            (loans_data["Health factor"] > 0)  # TODO: debug the negative HFs
+            & loans_data["Debt (USD)"].between(
+                debt_usd_lower_bound, debt_usd_upper_bound
+            )
+        ]
+        .sort_values("Health factor")
+        .iloc[:20],
+        use_container_width=True,
+    )
+
+
+def lottery(col1, streamlit, loans_data):
+    with col1:
+        user = streamlit.text_input("User")
+        protocol = streamlit.text_input("Protocol")
+
+        users_and_protocols_with_debt = list(
+            loans_data.loc[
+                loans_data["Debt (USD)"] > 0,
+                ["User", "Protocol"],
+            ].itertuples(index=False, name=None)
+        )
+
+        random_user, random_protocol = users_and_protocols_with_debt[
+            numpy.random.randint(len(users_and_protocols_with_debt))
+        ]
+
+        if not user:
+            streamlit.write(f"Selected random user = {random_user}.")
+            user = random_user
+        if not protocol:
+            streamlit.write(f"Selected random protocol = {random_protocol}.")
+            protocol = random_protocol
+
+    return user, protocol
+
+
+def get_pie_chart(col2, col3, streamlit, collateral_usd_amounts, debt_usd_amounts):
+    with col2:
+        figure = plotly.express.pie(
+            collateral_usd_amounts,
+            values="amount_usd",
+            names="token",
+            title="Collateral (USD)",
+            color_discrete_sequence=plotly.express.colors.sequential.Oranges_r,
+        )
+        streamlit.plotly_chart(figure, True)
+
+    with col3:
+        figure = plotly.express.pie(
+            debt_usd_amounts,
+            values="amount_usd",
+            names="token",
+            title="Debt (USD)",
+            color_discrete_sequence=plotly.express.colors.sequential.Greens_r,
+        )
+        streamlit.plotly_chart(figure, True)
+
+
+def get_parquets():
+    general_stats = pandas.read_parquet(
+        f"gs://{src.helpers.GS_BUCKET_NAME}/data/general_stats.parquet",
```

Comment:
`gs://{src.helpers.GS_BUCKET_NAME}/data/` this part is repeatable and can be moved to constant